Assets/MixedRealityToolkit/Utilities/Gltf/Serialization/GltfUtility.cs

```diff
@@ -0,0 +1,370 @@
+
+                gltfObject = GetGltfObjectFromJson(gltfJson);
+
+                if (gltfObject == null)
+                {
+                    Debug.LogError("Failed load Gltf Object from json schema.");
+                    return null;
+                }
+            }
+            else if (uri.Contains(".glb"))
+            {
+                isGlb = true;
+                byte[] glbData;
+
+                using (FileStream stream = File.Open(uri, FileMode.Open))
+                {
+                    glbData = new byte[stream.Length];
+
+                    if (loadAsynchronously)
+                    {
+                        await stream.ReadAsync(glbData, 0, (int)stream.Length);
+                    }
+                    else
+                    {
+                        stream.Read(glbData, 0, (int)stream.Length);
+                    }
+                }
+
+                gltfObject = GetGltfObjectFromGlb(glbData);
+
+                if (gltfObject == null)
+                {
+                    Debug.LogError("Failed to load GlTF Object from .glb!");
+                    return null;
+                }
+            }
+            else
+            {
+                Debug.LogError("Unsupported file name extension.");
+                return null;
+            }
+
+            gltfObject.Uri = uri;
+            int nameStart = uri.Replace("\\", "/").LastIndexOf("/", StringComparison.Ordinal) + 1;
+            int nameLength = uri.Length - nameStart;
+            gltfObject.Name = uri.Substring(nameStart, nameLength).Replace(isGlb ? ".glb" : ".gltf", string.Empty);
+
+            gltfObject.LoadAsynchronously = loadAsynchronously;
+            await gltfObject.ConstructAsync();
+
+            if (gltfObject.GameObjectReference == null)
+            {
+                Debug.LogError("Failed to construct Gltf Object.");
+            }
+
+            if (loadAsynchronously) { await Update; }
+
+            return gltfObject;
+        }
+
+        /// <summary>
+        /// Gets a glTF object from the provided json string.
+        /// </summary>
+        /// <param name="jsonString">String defining a glTF Object.</param>
+        /// <returns><see cref="GltfObject"/></returns>
+        /// <remarks>Returned <see cref="GltfObject"/> still needs to be initialized using <see cref="ConstructGltf.ConstructAsync"/>.</remarks>
+        public static GltfObject GetGltfObjectFromJson(string jsonString)
+        {
+            var gltfObject = JsonUtility.FromJson<GltfObject>(jsonString);
+
+            for (int i = 0; i < gltfObject.extensionsRequired?.Length; i++)
+            {
+                var extensionsRequired = GetGltfExtensionObjects(jsonString, gltfObject.extensionsRequired[i]);
+
+                foreach (var extensionRequired in extensionsRequired)
+                {
+                    // TODO Update this after KHR_materials_pbrSpecularGlossiness extension is supported
```

Comment:
is this a product of the gltf codebase this was based off of? I would prefer not to check in commented out code/we should probably Log warnings here if they aren't supported.

Reply:
warnings are logged below

Reply:
We should probably call out which specific todo items we want before checking in. Items required before checking in we should fix/uncomment. Items not required before checking in I would prefer to have stubbed out with warnings if at all calling out specific issues filed for the repo.